craft_providers/instance_config.py

```diff
@@ -19,7 +19,7 @@
 
 import io
 import pathlib
-from typing import Any, Dict, Optional
+from typing import Any, Optional
```

Comment:
suggestion: get rid of this last `Optional` use by changing the typing on the load method line 96 by 

```
    def load(
        cls,
        executor: Executor,
        config_path: pathlib.PurePath = pathlib.PurePath("/etc/craft-instance.conf"),
    ) -> "InstanceConfiguration | None":
```

?

Reply:
Changed it a bit further.